test/observable_list_test.dart

```diff
@@ -269,6 +269,16 @@ _runTests() {
       });
     });
 
+    test('sort or 2 elements from reaction fn', () {
+      ObservableList list = toObservable([3, 1]);
+      var futureSub = list.listChanges.listen(_reSort(list));
+      list.sort();
+      expect(list, [1, 3]);
+      // Delay clean up, because terminating listeners immediately hides bug.
```

Comment:
hmmm, I'm not sure about this approach. Delays tend to be arbitrary and/or slow down the tests.

Couldn't we do `return new Future(() { ... })` like the other tests do? That will cause the test to wait for the change delivery.

It is also possible to use deliverListChanges to force immediate delivery. The boolean return value is helpful as well:
https://github.com/dart-lang/observe/blob/636720f88ab93850dcebb981555722ee42dec684/lib/src/observable_list.dart#L259

For example, I think this would capture the bug while being a reliable test:
```dart
var list = toObservable([3, 1]);
// dummy listener to record changes
// TODO(jmesserly): should we just record changes always, to support the sync api?
list.changes.listen((records) => null);
list.sort();
expect(list.deliverListChanges(), true);
list.sort();
expect(list.deliverListChanges(), false);
list.sort();
expect(list.deliverListChanges(), false);
````